contracts/modules/staking/StakingNFT.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pragma solidity ^0.8.16;
+
+import "../../interfaces/IStakingNFT.sol";
+
+/// @dev Based on Solmate https://github.com/transmissions11/solmate/blob/main/src/tokens/ERC721.sol
+contract StakingNFT is IStakingNFT {
+
+  struct TokenInfo {
+    uint96 poolId;
+    address owner;
+  }
+
+  string public name;
+  string public symbol;
+  address public operator;
+  uint96 internal _totalSupply;
+
+  mapping(uint => TokenInfo) internal _tokenInfo;
+  mapping(address => uint) internal _balanceOf;
+  mapping(uint => address) public getApproved;
+  mapping(address => mapping(address => bool)) public isApprovedForAll;
+
+  constructor(
+    string memory _name,
+    string memory _symbol,
+    address _operator
+  ) {
+    name = _name;
+    symbol = _symbol;
+    operator = _operator;
+  }
+
+  // operator functions
+
+  function operatorTransferFrom(address from, address to, uint id) external {
+    require(msg.sender == operator, "NOT_OPERATOR");
+    require(from == ownerOf(id), "WRONG_FROM");
+    require(to != address(0), "INVALID_RECIPIENT");
+
+    // underflow of the sender's balance is impossible because we check for
+    // ownership above and the recipient's balance can't realistically overflow
+    unchecked {
+      _balanceOf[from]--;
+      _balanceOf[to]++;
+    }
+
+    _tokenInfo[id].owner = to;
+    delete getApproved[id];
+
+    emit Transfer(from, to, id);
+  }
+
+  function changeOperator(address newOperator) public {
+    require(msg.sender == operator, "NOT_OPERATOR");
+    require(newOperator != address(0), "INVALID_OPERATOR");
+    operator = newOperator;
+  }
+
+  // minting and supply
+
+  function mint(uint poolId, address to) public returns (uint id) {
+
+    // TODO: make sure msg.sender is a staking pool
+    require(to != address(0), "INVALID_RECIPIENT");
+
+    // counter overflow is incredibly unrealistic
+    unchecked {
+      id = _totalSupply++;
+      _balanceOf[to]++;
+    }
+
+    _tokenInfo[id].owner = to;
+    _tokenInfo[id].poolId = uint96(poolId);
+
+    emit Transfer(address(0), to, id);
+  }
+
+  function totalSupply() public view returns (uint) {
+    return _totalSupply;
+  }
+
+  // ERC165
+
+  function supportsInterface(bytes4 interfaceId) public pure returns (bool) {
+    return
+      interfaceId == 0x01ffc9a7 || // ERC165 Interface ID for ERC165
+      interfaceId == 0x80ac58cd || // ERC165 Interface ID for ERC721
+      interfaceId == 0x5b5e139f;   // ERC165 Interface ID for ERC721Metadata
+  }
+
+  // ERC721
+
+  function tokenURI(uint id) public view virtual returns (string memory) {
+    // TODO: implement token uri
```

Comment:
sounds like this should be addressed part of this PR

unless there's some product level decision we're waiting for, in which case let's link a ticket

Reply:
https://github.com/NexusMutual/smart-contracts/pull/613#discussion_r1065624351